docs/antrea-network-policy.md

```diff
@@ -1194,18 +1139,172 @@ spec:
         enableLogging: false
 ```
 
-In this example, the policy will be applied to all Pods whose ServiceAccount is in `ns-1` namespace and name as `sa-1`.
+In this example, the policy will be applied to all Pods whose ServiceAccount is `sa-1` of `ns-1`.
 Let's call those Pods "appliedToPods".
 The egress `to` section will select all Pods whose ServiceAccount is in `ns-2` namespace and name as `sa-2`.
 Let's call those Pods "egressPods".
+After this policy is applied, traffic from "appliedToPods" to "egressPods" will be dropped.
 
-So after this policy is applied, traffic from "appliedToPods" to "egressPods" will be dropped.
+Note: Antrea will use a reserved label key for internal processing `serviceAccount`.
+The reserved label looks like: `internal.antrea.io/service-account:[ServiceAccountName]`. Users should avoid using
+this label key in any entities no matter if a policy with `serviceAccount` is applied in the cluster.
 
-There is a CAVEAT after introducing `serviceAccount`:
+## ClusterGroup
 
-Antrea will use a reserved label key for internal processing `serviceAccount`.
-The reserved label looks like: `internal.antrea.io/service-account:[ServiceAccountName]`. Users should avoid using
-this label key in any entities whether a policy with `serviceAccount` is applied in the cluster.
+A ClusterGroup (CG) CRD is a specification of how workloads are grouped together.
+It allows admins to group Pods using traditional label selectors, which can then
+be referenced in ACNP in place of stand-alone `podSelector` and/or `namespaceSelector`.
+In addition to `podSelector` and `namespaceSelector`, ClusterGroup also supports the
+following ways to select endpoints:
+
+- Pod grouping by `serviceReference`. ClusterGroup specified by `serviceReference` will
+  contain the same Pod members that are currently selected by the Service's selector.
+- `ipBlock` or `ipBlocks` to share IPBlocks between ACNPs.
+- `childGroups` to select other ClusterGroups by name.
+
+ClusterGroups allow admins to separate the concern of grouping of workloads from
+the security aspect of Antrea-native policies.
+It adds another level of indirection allowing users to update group membership
+without having to update individual policy rules.
+
+### The ClusterGroup resource
+
+An example ClusterGroup might look like this:
+
+```yaml
+apiVersion: crd.antrea.io/v1alpha3
+kind: ClusterGroup
+metadata:
+  name: test-cg-sel
+spec:
+  podSelector:
+    matchLabels:
+      role: db
+  namespaceSelector:
+    matchLabels:
+      env: prod
+status:
+  conditions:
+    - type: "GroupMembersComputed"
+      status: "True"
+      lastTransitionTime: "2021-01-29T19:59:39Z"
```

Comment:
I'm wondering should we keep the `status` field in an example?

Reply:
what do you mean?

Reply:
I mean `status` is generated field. So maybe we don't need to put it in the example yaml? Like we didn't include `status` in ACNP example, etc.

Reply:
Make sense. Changed